black_it/utils/time_series.py

```diff
@@ -81,3 +84,45 @@ def get_mom_ts_1d(time_series: NDArray[np.float64]) -> NDArray[np.float64]:
     np.nan_to_num(avg_vec_mom, False)
 
     return avg_vec_mom
+
+
+def hp_filter(
+    time_series: NDArray[np.float64], lamb: float = 1600
+) -> Tuple[NDArray[np.float64], NDArray[np.float64]]:
+    """
+    Apply the HP filter to a time series.
+
+    Args:
+        time_series: the one dimensional series to be filtered
+        lamb: the Hodrick-Prescott smoothing parameter, a value of 1600 is suggested for quarterly data
+
+    Returns:
+        the cycle and the trend of the HP decomposition of the time series
+
+    References:
+        Hodrick, R.J, and E. C. Prescott. 1980. "Postwar U.S. Business Cycles: An Empirical Investigation."
+        Carnegie Mellon University discussion paper no. 451`.
```

Comment:
```suggestion
        Carnegie Mellon University discussion paper no. 451.
```

:slightly_smiling_face:

Reply:
Well spotted!!